src/server/common/helpers/fetch/fetch-buckets.js

```diff
@@ -0,0 +1,22 @@
+import { config } from '~/src/config/config.js'
+import { fetcher } from '~/src/server/common/helpers/fetch/fetcher.js'
+
+async function fetchBuckets(environment, serviceName, logger) {
+  try {
+    const endpoint =
+      config.get('portalBackendUrl') +
+      `/tenant-buckets/${serviceName}/${environment}`
+    const { data } = await fetcher(endpoint)
+
+    return data
+  } catch (error) {
+    // We are catching here because a 404 can be thrown when a service has not had buckets set up on an environment
```

Comment:
There may be other errors than missing bucket?

Reply:
True, but I must admit this is a copy/paste of `fetch-secrets.js` which does the same... maybe @feedmypixel can comment?

Reply:
So this is a bit of an anti-pattern we have because the `fetchers` throw on anything not in the 200 range. I have a ticket to overhaul all of the `fetch` stuff and have something rather nice in the wings for it. I'd keep this as is as its the same as elsewhere in the codebase.

@flurdy looking at the PBE we can either have an ok or 404 from these endpoints - https://github.com/DEFRA/cdp-portal-backend/blob/e444a7b64631b289654a4268a82aa6bb30e7e601/Defra.Cdp.Backend.Api/Endpoints/TenantBucketsEndpoint.cs

Reply:
Yeah fair enough.